### PR TITLE
LPS-146012 [CP 2.0] Fix subscriptions navbar to show only products with subscriptions

### PR DESCRIPTION
**Pull request by claricebrasil**

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js [hunk 1]
```diff
@@ -34,73 +34,71 @@ const Overview = ({project, subscriptionGroups}) => {
 		SUBSCRIPTIONS_STATUS.future,
 	]);
 
+	const [allSubscriptions, setAllSubscriptions] = useState([]);
+
+	const [
+		subscriptionGroupsWithSubscriptions,
+		setSubscriptionGroupsWithSubscriptions,
+	] = useState([]);
+
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js:43]:
```suggestion
	const [subscriptionsGroups, setSubscriptionsGroups] = useState([]);

```

Let's reduce the number of states, I saw that the allSubscription state is only used in useEffect

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js [hunk 2]
```diff
@@ -34,73 +34,71 @@ const Overview = ({project, subscriptionGroups}) => {
...
+								parseAccountSubscriptionGroupERC(
+									subscriptionGroup.name
+								)
+						)
+				);
+
+				setAllSubscriptions(dataAllSubscriptions);
+
+				setSubscriptionGroupsWithSubscriptions(
+					accountSubscriptionGroups
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js:95]: Convert this useEffect to an useMemo. return an object with the accountSubscriptionGroups and the dataAllsubscriptions
```suggestion
			return {
			accountSubscriptionGroups,
			allSubscriptions: dataAllSubscriptions
			}
```

Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js:95]:
```suggestion

```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js [hunk 3]
```diff
@@ -34,73 +34,71 @@ const Overview = ({project, subscriptionGroups}) => {
...
-			previousValue +
-			` subscriptionStatus eq '${currentValue}' or accountSubscriptionGroupERC eq '${
-				project.accountKey
-			}_${parseAccountSubscriptionGroupERC(
-				selectedSubscriptionGroup
-			)}' and`
-		);
-	};
+	useEffect(() => {
+		const subscriptionsToRender = allSubscriptions
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js:49]: Use here the useMemo variable suggested below, example:
```suggestion
		const subscriptionsCards = accountSubscriptionsData?.allSubscriptions
```
Also, I've renamed to subscriptionsCards, what do you think?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js [hunk 4]
```diff
@@ -34,73 +34,71 @@ const Overview = ({project, subscriptionGroups}) => {
...
+						`${project.accountKey}_`,
+						''
+					) ===
+					parseAccountSubscriptionGroupERC(selectedSubscriptionGroup)
+			)
+			.filter((subscription) =>
+				selectedStatus.includes(subscription.subscriptionStatus)
+			);
+
+		setAccountSubscriptions(subscriptionsToRender);
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js:62]:
```suggestion
		setSubscriptionsGroups(subscriptionsCards);
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js [hunk 5]
```diff
@@ -117,41 +115,57 @@ const Overview = ({project, subscriptionGroups}) => {
...
-							cardSubscriptionData={accountSubscription}
-							key={index}
+			{!!subscriptionGroupsWithSubscriptions.length && (
+				<>
+					<div
+						className={classNames('align-items-center d-flex', {
+							'justify-content-between':
+								subscriptionGroupsWithSubscriptions.length < 5,
+							'justify-content-evenly':
+								subscriptionGroupsWithSubscriptions.length > 4,
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Overview/index.js:125]:
```suggestion
								accountSubscriptionsData?.accountSubscriptionGroups.length < 5,
							'justify-content-evenly':
								accountSubscriptionsData?.accountSubscriptionGroups.length > 4,
```

**Review by iagochaves — CHANGES_REQUESTED** [review]

**Comment by jairmedeiros** [comment]
LGTM 🔥

**Comment by jairmedeiros** [comment]
@kevenleone

**Comment by jairmedeiros** [comment]
ci:test:sf

**Comment by jairmedeiros** [comment]
ci:test:stable

**Comment by jairmedeiros** [comment]
ci:test:relevant

**Review by kevenleone — APPROVED** [review]
LGTM 🔥

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**